scripts/ci/check_compliance.py

```diff
@@ -1541,20 +1544,40 @@ class KeepSorted(ComplianceTest):
 
     MARKER = "zephyr-keep-sorted"
 
-    def block_is_sorted(self, block_data):
-        lines = []
+    def block_check_sorted(self, block_data, regex):
+        def _test_indent(txt: str):
+            return txt.startswith((" ", "\t"))
+
+        if regex is None:
+            block_data = textwrap.dedent(block_data)
 
-        for line in textwrap.dedent(block_data).splitlines():
-            if len(lines) > 0 and line.startswith((" ", "\t")):
-                # Fold back indented lines
-                lines[-1] += line.strip()
+        lines = block_data.splitlines()
+        last = ''
+
+        for idx, line in enumerate(lines):
+            if not line.strip():
+                # Ignore blank lines
+                continue
+
+            if regex:
+                # check for regex
+                if not re.match(regex, line):
+                    continue
             else:
-                lines.append(line.strip())
+                # Ignore comments for many file types (CMake, python, Kconfig, ...)
+                if _test_indent(line) or line.startswith("#"):
```

Comment:
I ported the comment check from the old version, but I'm actually starting to think that this can hide issues, for example, the following won't complain now:

```c
/* zephyr-keep-sorted-start */
#define OPTION_B 0
#define OPTION_A 1
/* zephyr-keep-sorted-stop */
```

Reply:
Also slashes were a very bad choice to find a regex pattern 😆 need to fix.

Reply:
I've removed the comment check, as it's not used in tree.